climada/hazard/centroids/centr.py

```diff
@@ -491,52 +491,83 @@ def clear(self):
-            right = max(self.total_bounds[2], centr.total_bounds[2])
-            top = max(self.total_bounds[3], centr.total_bounds[3])
-            crs = self.meta['crs']
-            width = (right - left) / self.meta['transform'][0]
-            height = (bottom - top) / self.meta['transform'][4]
-            self.meta = {
-                'dtype': 'float32',
-                'width': width,
-                'height': height,
-                'crs': crs,
-                'transform': rasterio.Affine(self.meta['transform'][0], 0.0, left,
-                                             0.0, self.meta['transform'][4], top),
-            }
-            self.lat, self.lon = np.array([]), np.array([])
+
+        if self.meta and not self.lat.any():
+            self.set_meta_to_lat_lon()
+        if centr.meta and not centr.lat.any():
+            centr.set_meta_to_lat_lon()
+
+        if not u_coord.equal_crs(centr.geometry.crs, self.geometry.crs):
+            raise ValueError('Different CRS not accepted.')
+
+        self.meta = dict()
+        self.__dict__.update(self.union(centr).__dict__)
+
+
+    def union(self, *others):
+        """
+        Create the union of centroids from the inputs.
+
+        The centroids are combined together point by point.
+        Rasters are converted to points and raster information is lost.
+        All centroids must have the same CRS.
+
+        Parameters
+        ----------
+        others : any number of climada.hazard.Centroids()
+            Centroids to form the union with
+
+        Returns
+        -------
+        centroids : climada.hazard.Centroids()
+            Centroids containing the union of the centroids in centroids_list.
+
+        """
+
+        if self.lat.any() or self.meta:
+            centroids = copy.deepcopy(self)
         else:
-            LOGGER.debug('Appending points')
-            if not u_coord.equal_crs(centr.geometry.crs, self.geometry.crs):
-                raise ValueError('Different CRS not accepted.')
-            self.lat = np.append(self.lat, centr.lat)
-            self.lon = np.append(self.lon, centr.lon)
-            self.meta = dict()
-
-        # append all 1-dim variables
-        for (var_name, var_val), centr_val in zip(self.__dict__.items(),
-                                                  centr.__dict__.values()):
-            if isinstance(var_val, np.ndarray) and var_val.ndim == 1 and \
-            var_name not in ('lat', 'lon'):
-                setattr(self, var_name, np.append(var_val, centr_val).
-                        astype(var_val.dtype, copy=False))
+            centroids = copy.deepcopy(others[0])
+            others = others[1:]
+        if centroids.meta and not centroids.lat.any():
+            centroids.set_meta_to_lat_lon()
+
+        centroids.set_geometry_points()
+        for cent in others:
+            if cent.meta and not cent.lat.any():
+                cent.set_meta_to_lat_lon()
+            cent.set_geometry_points()
```

Comment:
I think we should use `set_geometry_points` only if at least one of the centroids in the list has a `geometry` attribute set.